File: python/day76.py
import getpass

# 1. Securely prompt for password (hides text while typing)
password = getpass.getpass("Enter your password: ")

# 2. Check for an uppercase letter without Regex
has_uppercase = False

for char in password:
    # .isupper() returns True only if the character is a letter and uppercase
    if char.isupper():
        has_uppercase = True
        break  # Exit loop early once found

# 3. Output the result
if has_uppercase:
    print("Password valid: Contains an uppercase letter.")
else:
    print("Password invalid: Must contain at least one uppercase letter.")

File: js/day76.js
const readline = require('readline');

// 1. Create the interface for terminal input
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// 2. Ask the user for the password
rl.question('Enter your password: ', (password) => {
  let hasUppercase = false;

  // 3. Loop through characters to check for uppercase letters
  for (let i = 0; i < password.length; i++) {
    const char = password[i];
    
    // Check if the character is between A and Z
    if (char >= 'A' && char <= 'Z') {
      hasUppercase = true;
      break; 
    }
  }

  // 4. Output result and close the interface
  if (hasUppercase) {
    console.log('Password valid: Contains an uppercase letter.');
  } else {
    console.log('Password invalid: Must contain at least one uppercase letter.');
  }

  rl.close();
});
